tfx_bsl/beam/run_inference.py

```diff
@@ -349,16 +527,31 @@ class _RemotePredictDoFn(_BaseDoFn):
   without having access to cloud-hosted model's signatures.
   """
 
-  def __init__(self, inference_spec_type: model_spec_pb2.InferenceSpecType,
-               pipeline_options: PipelineOptions):
-    super(_RemotePredictDoFn, self).__init__(inference_spec_type)
+  def __init__(
+      self,
+      pipeline_options: PipelineOptions,
+      fixed_inference_spec_type: model_spec_pb2.InferenceSpecType = None
+  ):
+    super(_RemotePredictDoFn, self).__init__(fixed_inference_spec_type)
+    self._pipeline_options = pipeline_options
+    self._fixed_inference_spec_type = fixed_inference_spec_type
+    self._use_fixed_model = (self._fixed_inference_spec_type is not None)
+
+  def setup(self):
+    super(_RemotePredictDoFn, self).setup()
+    if self._use_fixed_model:
+      self._setup_model(self._fixed_inference_spec_type)
```

Comment:
Why is this in setup instead of __init__ ?

Reply:
Moving this to `__init__` would require the api client to be serializable (as part of the DoFn). This may be possible, but may also lead to some strange issues if multiple DoFn's sharing an api client is a problem. The original code configured this in `setup` so this PR is just maintaining that convention.